backend/src/v5/models/users.js

```diff
@@ -207,4 +209,35 @@ User.getAvatar = async (username) => {
 
 User.uploadAvatar = (username, avatarBuffer) => updateUser(username, { $set: { 'customData.avatar': { data: avatarBuffer } } });
 
+User.resetPasswordToken = async (username) => {
+	let email; let resetPasswordToken; let
+		firstName;
+
+	try {
+		const { customData } = await User.getUserByQuery({ user: username }, { user: 1, 'customData.email': 1, 'customData.firstName': 1 });
+		email = customData.email;
+		firstName = customData.firstName;
+
+		const expiryAt = new Date();
+		expiryAt.setHours(expiryAt.getHours() + config.tokenExpiry.forgotPassword);
+
+		resetPasswordToken = {
+			token: generateHashString(64),
```

Comment:
default is fine, there's no reason for it to be 64 characters long.

```suggestion
			token: generateHashString(),
```